### PR TITLE
[AdminApi] Fix command

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | -----
| Branch?         | 1.2
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/AdminApiBundle/Command/CreateClientCommand.php [hunk 1]
```diff
@@ -15,13 +15,25 @@
...
 
-final class CreateClientCommand extends ContainerAwareCommand
+final class CreateClientCommand extends Command
 {
+    /**
+     * @var ClientManagerInterface
+     */
+    private $clientManager;
+
+    public function __construct(string $name = null, ClientManagerInterface $clientManager)
```
Comment (pamil) [inline src/Sylius/Bundle/AdminApiBundle/Command/CreateClientCommand.php]: Unfortunately, adding a mandatory argument to the constructor method is considered a BC break.

Comment (lchrusciel) [reply src/Sylius/Bundle/AdminApiBundle/Command/CreateClientCommand.php]: I can add a separate class and override service definition in the configuration if current Symfony version is 4.1. 

This class will be deprecated then.

Comment (lchrusciel) [reply src/Sylius/Bundle/AdminApiBundle/Command/CreateClientCommand.php]: What about this solution?

File: src/Sylius/Bundle/AdminApiBundle/Resources/config/services.xml [hunk 2]
```diff
@@ -23,6 +23,8 @@
         <defaults public="true" />
 
         <service id="Sylius\Bundle\AdminApiBundle\Command\CreateClientCommand">
+            <argument />
```
Comment (pamil) [inline src/Sylius/Bundle/AdminApiBundle/Resources/config/services.xml]: This passes an empty string now.

Comment (pamil) [reply src/Sylius/Bundle/AdminApiBundle/Resources/config/services.xml]: `<argument>null</argument>` is the solution for passing null.

Comment (lchrusciel) [reply src/Sylius/Bundle/AdminApiBundle/Resources/config/services.xml]: Won't it just pass `null` as a string?

Comment (pamil) [reply src/Sylius/Bundle/AdminApiBundle/Resources/config/services.xml]: Nope, I've checked it myself, writing an integration test for that would be 11/10 though :)

File: src/Sylius/Bundle/AdminApiBundle/Resources/config/services.xml [hunk 3]
```diff
@@ -23,6 +23,8 @@
         <defaults public="true" />
 
         <service id="Sylius\Bundle\AdminApiBundle\Command\CreateClientCommand">
+            <argument on-invalid="null"/>
```
Comment (pamil) [inline src/Sylius/Bundle/AdminApiBundle/Resources/config/services.xml]: It still passes an empty string as command name which causes `InvalidArgumentException` to be thrown. What was wrong with `<argument>null</argument>`?

**Comment by pamil** [comment]
Thank you, Łukasz! 🎉

**Pull request merged**